File: aoc2023/test/aoc2023/day04_test.clj
(ns aoc2023.day04-test
  (:require [clojure.test :refer :all]
            [clojure.java.io]
            [aoc2023.day04 :as day04]))

(def sample-input (slurp (clojure.java.io/resource "day04-sample.txt")))

(deftest test-part1
  (testing "Day 4 Part 1"
    (is (= 13 (day04/part1 sample-input)))
  )
)

;; (deftest test-part2
;;   (testing "Day 3 Part 2"
;;     (is (= 2286 (day01/part2 sample-input)))
;;   )
;; )

File: 2024/day04/day04_test.go
package day04

import (
	"fmt"
    "testing"
    )

var sample = "sample.txt"

func TestPart1(t *testing.T) {
  result := Part1(sample)
  if result != 18 {
    fmt.Println(result)
  	t.Fail()
  }
}

func TestPart2(t *testing.T) {
  result := Part2(sample)
  if result != 9 {
    fmt.Println(result)
  	t.Fail()
  }
}
